MyCalculator.c

```diff
@@ -0,0 +1,176 @@
+            int value = 0;
+            while (i < length && isdigit(expression[i])) {
+                value = value * 10 + (expression[i] - '0');
+                i++;
+            }
+            if (operand_top >= MAX_STACK_SIZE - 1) {
+                *error = ERROR_INVALID_EXPRESSION;
+                result = 0;
+                break;
+            }
+            operands[++operand_top] = value;
+        }
+        // Process operators
+        else if (expression[i] == '+' || expression[i] == '-' || expression[i] == '*' || expression[i] == '/') {
+            while (operator_top >= 0 && 
+                   operator_precedence(operators[operator_top]) >= operator_precedence(expression[i])) {
+                if (operand_top < 1) {
+                    *error = ERROR_INVALID_EXPRESSION;
+                    result = 0;
+                    break;
+                }
+                char op = operators[operator_top--];
+                int b = operands[operand_top--];
+                int a = operands[operand_top--];
+                operands[++operand_top] = apply_operator(op, a, b, error);
+                if (*error) {
+                    result = 0;
+                    break;
+                }
+            }
+            operators[++operator_top] = expression[i];
+            i++;
+        }
+        // Invalid character
+        else {
+            *error = ERROR_INVALID_EXPRESSION;
+            result = 0;
+            break;
+        }
+    }
+
+    // Process remaining operators
+    while (*error == 0 && operator_top >= 0) {
+        if (operand_top < 1) {
+            *error = ERROR_INVALID_EXPRESSION;
+            result = 0;
+            break;
+        }
+        char op = operators[operator_top--];
+        int b = operands[operand_top--];
+        int a = operands[operand_top--];
+        operands[++operand_top] = apply_operator(op, a, b, error);
+        if (*error) {
+            result = 0;
+            break;
+        }
+    }
+
+    if (*error == 0) {
+        result = operands[operand_top];
+    }
+
+    return result;
+}
+
+// Main function
+// Main function
+int main() {
+    char expression[1000];
+    char choice;
+
+    do {
+        printf("Enter the expression: ");
+        fgets(expression, sizeof(expression), stdin);
+
+        // Remove newline character if present
+        for (int i = 0; expression[i] != '\0'; i++) {
+            if (expression[i] == '\n') {
+                expression[i] = '\0';
+                break;
+            }
+        }
+
+        int error = 0;
+        int result = evaluate_expression(expression, &error);
+
+        if (error == ERROR_INVALID_EXPRESSION) {
+            printf("Error: Invalid expression.\n");
+        } else if (error == ERROR_DIVISION_BY_ZERO) {
+            printf("Error: Division by zero.\n");
+        } else if (error == ERROR_UNKNOWN) {
+            printf("Error: Unknown error.\n");
+        } else {
+            printf("Result: %d\n", result);
+        }
```

Comment:
Have proper formatting, like this:
if (condition)
{
    block of code
}
else if (condition)
{
    block of code
}
else
{
    block of code
}